### ISSUE TITLE
yamlfmt insert blank line between commented line

### ISSUE THREAD

**Issue by sharuzzaman** [google/yamlfmt#34]
I have the following YAML:

```
---
Resources:
  ElasticBeanstalk:
    Type: AWS::CloudFormation::Stack
    Properties:
      #NotificationARNs: 
      #  - String
      #Parameters: 
      #  Key : Value
      #Tags: 
      #  - Tag
      TemplateURL: https://examplebucket.s3.ap-southeast-1.amazonaws.com/example-file.json
      TimeoutInMinutes: 5
```

after formatting, each commented line will have a blank line inserted, becoming:

```
Resources:
  ElasticBeanstalk:
    Type: AWS::CloudFormation::Stack
    Properties:
      #NotificationARNs: 

      #  - String

      #Parameters: 

      #  Key : Value

      #Tags: 

      #  - Tag
      TemplateURL: https://examplebucket.s3.ap-southeast-1.amazonaws.com/example-file.json
      TimeoutInMinutes: 5
```

I don't see any option to configure the formatter so that it does not include a blank line after each comment.

Thank you.

**Comment by braydonk** [comment]
Hi @sharuzzaman,

Unfortunately at this time there aren't a ton of configuration options, so you wouldn't be able to configure a behaviour like this for the time being.

However, I'm unable to reproduce this behaviour locally at the moment. I tried copying the yaml from the first block into a file on its own and it did not insert the newlines as you've mentioned. 

x.yaml: 
```yaml
---
Resources:
  ElasticBeanstalk:
    Type: AWS::CloudFormation::Stack
    Properties:
      #NotificationARNs: 
      #  - String
      #Parameters: 
      #  Key : Value
      #Tags: 
      #  - Tag
      TemplateURL: https://examplebucket.s3.ap-southeast-1.amazonaws.com/example-file.json
      TimeoutInMinutes: 5
```
Piping it into `stdin`:
```bash
braydonk@bk:~/Git/yamlfmt$ cat x.yaml | yamlfmt -dry
dry run produced no output
```

There's a chance this is a platform issue. What operating system are you on?

**Comment by braydonk** [comment]
This seems to be a Windows-specific issue. Gotta be some kind of CRLF problem if I had to guess. I'll keep looking into it.

**Comment by braydonk** [comment]
This is tough. It's a combination of 2 bugs in `yaml.v3`, one is pretty simple and one is not.

`yaml.Encoder` always writes `\n`, I'll make a PR upstream that makes it configurable to write `\r\n` instead

When the parser scans a head comment ending in `\r\n`, if the next line is anything but another line break, it inserts an extra line. This is deep in the parser code, and won't be fast to solve...

The silver lining is that this is not a universal Windows issue, this is particularly with CRLF line endings. That is a possible workaround if you are flexible with line endings in your project. Regardless, I'll do what I can to get this solved.

**Comment by braydonk** [comment]
Opened https://github.com/go-yaml/yaml/issues/891 and a PR for it https://github.com/go-yaml/yaml/pull/892

**Comment by braydonk** [comment]
I managed to find the problem in the parser code too. Opened https://github.com/go-yaml/yaml/issues/893 and https://github.com/go-yaml/yaml/pull/894

**Comment by braydonk** [comment]
I've opened a hotfix for the behaviour while waiting for the upstream stuff to get fixed, and that's what will go in v0.3.0 so that Windows users can be fixed sooner.

**Comment by sharuzzaman** [comment]
I would like to update that this issue is not happening again and the hotfix is working correctly. Thank you
